src/Playwright/Core/BrowserContext.cs

```diff
@@ -324,34 +340,28 @@ public Task<IPage> RunAndWaitForPageAsync(Func<Task> action, BrowserContextRunAn
 
         public void SetDefaultTimeout(float timeout) => DefaultTimeout = timeout;
 
-        internal void OnRoute(Route route, IRequest request)
+        internal async Task OnRouteAsync(Route route, IRequest request)
         {
-            foreach (var routeHandler in _routes)
+            var routeHandlers = _routes.Where(r => r.Regex?.IsMatch(request.Url) == true ||
```

Comment:
This needs a small tweaking according to the latest upstream change (https://github.com/microsoft/playwright/commit/ae6f48c4b8bb55db693dfdfd27592d9206e0d46d)